### PR TITLE
Rename namespaces

### PR DESCRIPTION
**Pull request by cliu123**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Maintenance
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__

 4. __Why these changes are required?__ 
 Rename namespace to follow OpenSearch naming convention

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)
 No changes in behavior

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 UT
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by hsiang9431-amzn — DISMISSED** [review]

File: THIRD-PARTY.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ Lists of 69 third-party dependencies.
      (The Apache Software License, Version 2.0) Jackson dataformat: CBOR (com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.8.10 - http://github.com/FasterXML/jackson-dataformats-binary)
      (The Apache Software License, Version 2.0) Jackson dataformat: Smile (com.fasterxml.jackson.dataformat:jackson-dataformat-smile:2.8.10 - http://github.com/FasterXML/jackson-dataformats-binary)
      (The Apache Software License, Version 2.0) Jackson-dataformat-YAML (com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.8.10 - https://github.com/FasterXML/jackson)
-     (The Apache Software License, Version 2.0) OpenDistro Security SSL (com.amazon.opendistroforelasticsearch:opendistro-elasticsearch-security-ssl:0.0.8.0 - https://github.com/opendistro-for-elasticsearch/security-ssl)
+     (The Apache Software License, Version 2.0) OpenDistro Security SSL (org.opensearch:opendistro-elasticsearch-security-ssl:0.0.8.0 - https://github.com/opendistro-for-elasticsearch/security-ssl)
```
Comment (vrozov) [inline THIRD-PARTY.txt:8]: this artifact does not exist

File: plugin-descriptor.properties [hunk 2]
```diff
@@ -9,7 +9,7 @@ version=1.0.0.0-beta1
 name=opensearch-security
 #
 # 'classname': the name of the class to load, fully-qualified.
-classname=com.amazon.opendistroforelasticsearch.security.OpenDistroSecurityPlugin
+classname=org.opensearch.security.OpenDistroSecurityPlugin
```
Comment (vrozov) [inline plugin-descriptor.properties:12]: package rename needs to be combined with class renames.

File: src/main/java/org/opensearch/security/OpenDistroSecurityPlugin.java [hunk 3]
```diff
@@ -544,7 +543,7 @@ public void onPreQueryPhase(SearchContext context) {
                 @Override
                 public void onNewReaderContext(ReaderContext readerContext) {
                     final boolean interClusterRequest = HeaderHelper.isInterClusterRequest(threadPool.getThreadContext());
-                    if (Origin.LOCAL.toString().equals(threadPool.getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN))
+                    if (AuditLog.Origin.LOCAL.toString().equals(threadPool.getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN))
```
Comment (vrozov) [inline src/main/java/org/opensearch/security/OpenDistroSecurityPlugin.java:546]: Undo

File: src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java [hunk 4]
```diff
@@ -13,20 +13,20 @@
...
-import com.amazon.opendistroforelasticsearch.security.configuration.ConfigurationRepository;
-import com.amazon.opendistroforelasticsearch.security.dlic.rest.validation.AbstractConfigurationValidator;
-import com.amazon.opendistroforelasticsearch.security.dlic.rest.validation.InternalUsersValidator;
-import com.amazon.opendistroforelasticsearch.security.privileges.PrivilegesEvaluator;
-import com.amazon.opendistroforelasticsearch.security.securityconf.Hashed;
-import com.amazon.opendistroforelasticsearch.security.securityconf.impl.CType;
-import com.amazon.opendistroforelasticsearch.security.securityconf.impl.SecurityDynamicConfiguration;
-import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
-import com.amazon.opendistroforelasticsearch.security.support.SecurityJsonNode;
+package org.opensearch.security.dlic.rest.api;
```
Comment (vrozov) [inline src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java:16]: Why some packages in `dlic` changes and some are not?

Comment (cliu123) [reply src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java:16]: Considering the third-party dependencies such as ldap, the `com.amazon.dlic` is not changed. Only changed the package of security plugin `com.amazon.opendistroforelasticsearch.security`

File: THIRD-PARTY.txt (hunk 1 above)
Comment (cliu123) [reply THIRD-PARTY.txt]: Reverted it back

File: src/main/java/org/opensearch/security/OpenDistroSecurityPlugin.java (hunk 3 above)
Comment (cliu123) [reply src/main/java/org/opensearch/security/OpenDistroSecurityPlugin.java]: Done

**Pull request closed without merging**